raw_linux.go

```diff
@@ -279,6 +279,34 @@ func (p *packetConn) SetBPF(filter []bpf.RawInstruction) error {
 	)
 }
 
+// SetPromisc enables/disables interface promiscuous mode through syscall
+// when thread exits interface returns to previous state
+// 0 == disabled, 1 == enabled
+type packet_mreq struct {
```

Comment:
CL submitted: https://go-review.googlesource.com/c/51710/.

Once this is merged, we can pull latest `x/sys/unix` and use this structure directly.